src/main/java/uk/gov/justice/digital/delius/controller/secure/StaffResource.java

```diff
@@ -99,4 +121,65 @@ public StaffDetails getStaffDetailsForUsername(
         @PageableDefault(sort = {"secondName", "firstName"}, direction = Direction.ASC, size = Integer.MAX_VALUE) final Pageable pageable) {
         return staffService.getManageSupervisionsEligibleOffendersByUsername(username, pageable);
     }
+
+	@ApiOperation(
+		value = "Return the full caseload for a probation staff/officer",
+		notes = "Currently, this endpoint is restricted to offender managers and order supervisors. Additional management types (e.g. requirements, reports) may be added later.")
+	@GetMapping(path = "/staff/staffIdentifier/{staffIdentifier}/caseload")
+	public Caseload getCaseloadForStaff(
+		@ApiParam(name = "staffIdentifier", value = "Delius staff/officer identifier", example = "123456", required = true)
+		@NotNull @PathVariable(value = "staffIdentifier") final Long staffIdentifier) {
+		return caseloadService.getCaseloadByStaffIdentifier(staffIdentifier, OFFENDER_MANAGER, ORDER_SUPERVISOR)
+			.orElseThrow(() -> new NotFoundException(String.format("Staff member with identifier %d", staffIdentifier)));
+	}
+
+	@ApiOperation(value = "Return the full caseload for a probation staff/officer, returning only the managed offenders")
+	@GetMapping(path = "/staff/staffIdentifier/{staffIdentifier}/caseload/managedOffenders")
+	public Set<ManagedOffenderCrn> getCaseloadOffendersForStaff(
+		@ApiParam(name = "staffIdentifier", value = "Delius staff/officer identifier", example = "123456", required = true)
+		@NotNull @PathVariable(value = "staffIdentifier") final Long staffIdentifier) {
+		return caseloadService.getCaseloadByStaffIdentifier(staffIdentifier, OFFENDER_MANAGER)
+			.map(Caseload::getManagedOffenders)
+			.orElseThrow(() -> new NotFoundException(String.format("Staff member with identifier %d", staffIdentifier)));
+	}
+
+	@ApiOperation(value = "Return the full caseload for a probation staff/officer, returning only the supervised orders/events")
+	@GetMapping(path = "/staff/staffIdentifier/{staffIdentifier}/caseload/supervisedOrders")
```

Comment:
Yeah it's a strange one, as there are multiple ways to identify the resource.  I agree it may not be the pure REST way to do it, but I've stuck with this pattern as it's consistent with the rest of the Community API endpoints, as well as with a few other HMPPS APIs (e.g. assessments-api).

See for example, the Offender resource which can be identified by `crn` or `nomsNumber`: https://community-api-public.dev.probation.service.justice.gov.uk/swagger-ui/index.html#/Core%20offender 
or the other existing Staff resource endpoints, which can be referenced by `staffIdentifier` or `username`: https://community-api-public.dev.probation.service.justice.gov.uk/swagger-ui/index.html#/Staff

Reply:
I kind of disagree on this , as you can see on the previous endpoints, the id is in the path whne has not relation with the resource structure. as you mention crn : offenders/crn/{crn} and kind of make sense . But if the structure is clear like /secure/offenders/crn/{crn}/convictions/{convictionId}/nsis/{nsiId} for conviction you do not write convictions/convictionid/{convictionid}

Reply:
That's because there's only a single way to identify convictions.  However there are multiple ways to identify staff, teams, and offenders - which is why this pattern was needed (e.g. for offender by crn or nomsNumber in your example).